File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/communication/message/request/RequestFactory.java
package lunalauf.rms.utilities.network.communication.message.request;

import java.util.Random;

public class RequestFactory {

    private long messageIdCounter;

    public RequestFactory() {
        messageIdCounter = new Random().nextLong();
    }

    private long getNextMessageId() {
        messageIdCounter = (messageIdCounter + 1) % (Long.MAX_VALUE - 1);
        return messageIdCounter;
    }

    public MinigameRecordRequest createMinigameRecordRequest(long runnerId, int minigameId, int points) {
        MinigameRecordRequest request = new MinigameRecordRequest();
        request.messageId = getNextMessageId();
        request.runnerId = runnerId;
        request.minigameId = minigameId;
        request.points = points;
        return request;
    }

    public RoundCountRequest createRoundCoundRequest(long runnerId) {
        RoundCountRequest request = new RoundCountRequest();
        request.messageId = getNextMessageId();
        request.runnerId = runnerId;
        return request;
    }

    public RunnerInfoRequest createRunnerInfoRequest(long runnerId) {
        RunnerInfoRequest request = new RunnerInfoRequest();
        request.messageId = getNextMessageId();
        request.runnerId = runnerId;
        return request;
    }

}

File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/communication/message/request/RequestFactory.kt
package lunalauf.rms.utilities.network.communication.message.request

import java.util.*

class RequestFactory {
    private var messageIdCounter = Random().nextLong()

    private val nextMessageId: Long
        get() {
            messageIdCounter = (messageIdCounter + 1) % (Long.MAX_VALUE - 1)
            return messageIdCounter
        }

    fun createMinigameRecordRequest(runnerId: Long, minigameId: Int, points: Int): MinigameRecordRequest {
        val request = MinigameRecordRequest()
        request.messageId = nextMessageId
        request.runnerId = runnerId
        request.minigameId = minigameId
        request.points = points
        return request
    }

    fun createRoundCountRequest(runnerId: Long): RoundCountRequest {
        val request = RoundCountRequest()
        request.messageId = nextMessageId
        request.runnerId = runnerId
        return request
    }

    fun createRunnerInfoRequest(runnerId: Long): RunnerInfoRequest {
        val request = RunnerInfoRequest()
        request.messageId = nextMessageId
        request.runnerId = runnerId
        return request
    }
}
